CI: integrate-dm always fails

## Bug Report

<img width="853" height="194" alt="Image" src="https://github.com/user-attachments/assets/0420cf50-fa3d-4222-b459-b6f0f4c605f8" />

Definition: https://github.com/pingcap/tiup/blob/master/.github/workflows/integrate-dm.yaml

Might be TLS related:
```
 ERROR 2026 (HY000): TLS/SSL error: self-signed certificate in certificate chain
./test_upgrade.sh:script[38 66] Failed with status 1: ./script/task/run.sh
Error: Process completed with exit code 1.
```

--- a/.github/workflows/integrate-dm.yaml
+++ b/.github/workflows/integrate-dm.yaml
@@ -40,16 +40,15 @@ jobs:
           - '--native-ssh --do-cases test_cmd'
           - '--native-ssh --do-cases test_upgrade'
     steps:
-      - name: Set up Go 1.24
-        uses: actions/setup-go@v4
-        with:
-          go-version: 1.24.x
-        id: go
-
       - name: Check out code into the Go module directory
-        uses: actions/checkout@v3
+        uses: actions/checkout@v5
         with:
           ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set up Go
+        uses: actions/setup-go@v6
+        with:
+          go-version-file: 'go.mod'
 
       - name: Build build_integration_test
         run: |
@@ -82,7 +81,7 @@ jobs:
       - name: Run test suite
         id: test
         run: |
-          # ensuere docker ssh is ready
+          # ensure docker ssh is ready
           sleep 4
           # should not use -it
           # ref: https://stackoverflow.com/questions/43099116/error-the-input-device-is-not-a-tty
@@ -90,7 +89,6 @@ jobs:
 
       - name: Collect component log
         if: ${{ failure() }}
-        # if: ${{ failure() }}
         run: |
           docker exec tiup-cluster-control bash -c 'mkdir -p /tiup-cluster/logs; [[ -d ~/.tiup/logs ]] && find ~/.tiup/logs -type f -name "*.log" -exec cp {} /tiup-cluster/logs \; || true'
 

--- a/docker/control/Dockerfile
+++ b/docker/control/Dockerfile
@@ -2,6 +2,7 @@ FROM golang:1.24
 
 WORKDIR /tiup-cluster
 COPY . /tiup-cluster/
+COPY my.cnf /root/.my.cnf
 
 # tiup-cluster dependencies
  RUN apt-get -y -q update && \

--- a/docker/control/my.cnf
+++ b/docker/control/my.cnf
@@ -1,0 +1,2 @@
+[mysql]
+ssl-verify-server-cert=false

--- a/docker/docker-compose.dm.yml.tpl
+++ b/docker/docker-compose.dm.yml.tpl
@@ -5,7 +5,7 @@ services:
     networks:
       tiops:
         ipv4_address: __IPPREFIX__.201
-    image: mysql:5.7
+    image: mysql:8.0
     command: --default-authentication-plugin=mysql_native_password --log-bin=/var/lib/mysql/mysql-bin.log --server-id=1 --binlog-format=ROW --gtid_mode=ON --enforce-gtid-consistency=true
     restart: always
     environment:
@@ -17,7 +17,7 @@ services:
     networks:
       tiops:
         ipv4_address: __IPPREFIX__.202
-    image: mysql:5.7
+    image: mysql:8.0
     command: --default-authentication-plugin=mysql_native_password --log-bin=/var/lib/mysql/mysql-bin.log --server-id=2 --binlog-format=ROW --gtid_mode=ON --enforce-gtid-consistency=true
     restart: always
     environment:
@@ -29,4 +29,4 @@ services:
     networks:
       tiops:
         ipv4_address: __IPPREFIX__.211
-    image: pingcap/tidb:v4.0.3
+    image: pingcap/tidb:v8.5.3

